RecoHI/HiEvtPlaneAlgos/src/HiEvtPlaneFlatProducer.cc

```diff
@@ -205,27 +305,30 @@ HiEvtPlaneFlatProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetu
   for(int i = 0; i<NumEPNames; i++) {
     ep[i]=0;
   }
-  for (EvtPlaneCollection::const_iterator rp = evtPlanes->begin();rp !=evtPlanes->end(); rp++) {
-    if(rp->angle() > -5) {
-      string baseName = rp->label();
-      for(int i = 0; i< NumEPNames; i++) {
-	if(EPNames[i].compare(baseName)==0) {
-	  double psiFlat = flat[i]->GetFlatPsi(rp->angle(),vzr_sell,bin);
-	  epang[i]=psiFlat;
-	  if(EPNames[i].compare(rp->label())==0) {	    
-	    if(storeNames_) ep[i]= new EvtPlane(psiFlat, rp->sumSin(), rp->sumCos(),rp->label().data());
-	    else ep[i]= new EvtPlane(psiFlat, rp->sumSin(), rp->sumCos(),"");
-	  } 
-	}
-      }
-    }    
+  int indx = 0;
+  for (EvtPlaneCollection::const_iterator rp = evtPlanes_->begin();rp !=evtPlanes_->end(); rp++) {
+	double angorig = rp->angle();
+	double s = rp->sumSin();
+	double c = rp->sumCos();
+	double w = rp->sumw();
+	uint m = rp->mult();
+	
+	double psiOffset = angorig;
+	if(useOffsetPsi_) psiOffset = flat[indx]->GetOffsetPsi(s,c,w,m,vzr_sell,bin);
```

Comment:
at the face value (GetOffsetPsi is actually changing the state of the flat[indx]) this part of the code violates the rule that results of event processing for EDProducer modules should be independent.
I have a suggestion here
https://github.com/slava77/cmssw/compare/CMSSW_7_5_0_pre4/sign549/pull9000-9022-mo...slava77:CMSSW_7_5_0_pre4/sign549/pull9000-9022-fixMt
please check and adopt it or provide another solution that leaves the "flat" array constant between the events (modulo the possible changes at the run boundaries)